src/bridge/MatrixHandler.ts

```diff
@@ -13,14 +13,15 @@ import { toIrcLowerCase } from "../irc/formatting";
 import { AdminRoomHandler } from "./AdminRoomHandler";
 import { MembershipQueue } from "../util/MembershipQueue";
 
-function reqHandler(req: BridgeRequest, promise: PromiseLike<unknown>) {
-    return promise.then(function(res) {
+async function reqHandler(req: BridgeRequest, promise: PromiseLike<unknown>) {
+    try {
+        const res = await promise;
         req.resolve(res);
-        return res;
-    }, function(err) {
+    }
+    catch (err) {
         req.reject(err);
         throw err;
-    });
+    }
```

Comment:
The promise `reqHandler` returns used to resolve in `res` but no longer does this.
The try block or the function block should end on `return res;`.

Reply:
fixed.